### DISCUSSION TITLE
"Specified rules folder does not exist" Error Help

### DISCUSSION THREAD

**Discussion by roman-tasi** [jertel/elastalert2#617] [Q&A] [answered]
I created a Docker container but it is now giving this error in the logs:

`[root@ELK-Stack ~]# docker logs b1b73d0542c90acc1ab7a9ef410fffbec8556b8228afaba49b263ea58b6f9ece/`

`Traceback (most recent call last):
  File "/usr/local/bin/elastalert", line 8, in <module>
    sys.exit(main())
  File "/usr/local/lib/python3.10/site-packages/elastalert/elastalert.py", line 2155, in main
    client = ElastAlerter(args)
  File "/usr/local/lib/python3.10/site-packages/elastalert/elastalert.py", line 128, in __init__
    self.rules = self.rules_loader.load(self.conf, self.args)
  File "/usr/local/lib/python3.10/site-packages/elastalert/loaders.py", line 159, in load
    rule_files = self.get_names(conf, use_rule)
  File "/usr/local/lib/python3.10/site-packages/elastalert/loaders.py", line 558, in get_names
    raise EAException('Specified rule_folder does not exist: %s ' % ruledir)
elastalert.util.EAException: Specified rule_folder does not exist: /etc/elastalert/elastalert_rules
[root@ELK-Stack ~]#
`

I am unsure how to fix it because my `elastalert.yaml ` has this field:
`rules_folder: /etc/elastalert/elastalert_rules`

Can anyone help me?

**Comment by jertel** [comment, marked as answer]
You need another `-v` param if you want to share rules located on the host filesystem with the docker container. I suggest following some tutorials on Docker so you're better able to get past these concepts.

**Reply by nsano-rururu** [reply]
elasticsearch has already been started with docker. The docker network name is "es_default". I will explain on the assumption that elastalert will participate there.

Create a rule directory and rule file in addition to config.yaml
Mount when docker container starts

```sh
config.yaml
rule
  a.yaml
  
```

config.yaml

```yaml
rules_folder: /opt/elastalert/rules

run_every:
  seconds: 10

buffer_time:
  minutes: 15

es_host: elasticsearch
es_port: 9200

writeback_index: elastalert_status

alert_time_limit:
  days: 2
```

a.yaml

```yaml
alert:
  - "slack"
alert_subject: "Test {} 123 aa☃"
alert_subject_args:
  - "message"
  - "@log_name"
alert_text: "Test {}  123 bb☃"
alert_text_args:
  - "message"
filter:
  - query:
      query_string:
        query: "@timestamp:*"
index: "mariadblog-*"
is_enabled: true
name: "a"
num_events: 2
realert:
  minutes: 5
slack_webhook_url: 'https://hooks.slack.com/services/xxxxxxxxx'
slack_channel_override: "#abc"
slack_emoji_override: ":kissing_cat:"
slack_msg_color: "warning"
slack_parse_override: "none"
slack_timeout: 10
slack_title_link: "http://127.0.0.1:8080/rules/a"
slack_username_override: "elastalert"
terms_size: 50
timeframe:
  minutes: 5
timestamp_field: "@timestamp"
timestamp_type: "iso"
type: "frequency"
use_strftime_index: false
```


```yaml
docker run --net=es_default -d --name elastalert --restart=always \
-v $(pwd)/config.yaml:/opt/elastalert/config.yaml \
-v $(pwd)/rule:/opt/elastalert/rules \
jertel/elastalert2:2.2.3 --verbose
```

```sh
$ docker logs -f elastalert

Reading Elastic 6 index mappings:
Reading index mapping 'es_mappings/6/silence.json'
Reading index mapping 'es_mappings/6/elastalert_status.json'
Reading index mapping 'es_mappings/6/elastalert.json'
Reading index mapping 'es_mappings/6/past_elastalert.json'
Reading index mapping 'es_mappings/6/elastalert_error.json'
Index elastalert_status already exists. Skipping index creation.
INFO:elastalert:Starting up
INFO:elastalert:Disabled rules are: []
INFO:elastalert:Sleeping for 9.99992 seconds
INFO:elastalert:Background configuration change check run at 2021-12-20 16:51 UTC
INFO:elastalert:Background alerts thread 0 pending alerts sent at 2021-12-20 16:51 UTC
INFO:elastalert:Disabled rules are: []
INFO:elastalert:Sleeping for 9.999901 seconds
INFO:elastalert:Queried rule a from 2021-12-20 16:51 UTC to 2021-12-20 16:51 UTC: 0 / 0 hits
INFO:elastalert:Ran a from 2021-12-20 16:51 UTC to 2021-12-20 16:51 UTC: 0 query hits (0 already seen), 0 matches, 0 alerts sent
INFO:elastalert:a range 29
INFO:elastalert:Background configuration change check run at 2021-12-20 16:52 UTC
INFO:elastalert:Background alerts thread 0 pending alerts sent at 2021-12-20 16:52 UTC
INFO:elastalert:Disabled rules are: []
INFO:elastalert:Sleeping for 9.999861 seconds
INFO:elastalert:Queried rule a from 2021-12-20 16:51 UTC to 2021-12-20 16:52 UTC: 0 / 0 hits
INFO:elastalert:Ran a from 2021-12-20 16:51 UTC to 2021-12-20 16:52 UTC: 0 query hits (0 already seen), 0 matches, 0 alerts sent
INFO:elastalert:a range 43
```

**Reply by nsano-rururu** [reply]
To be honest, the description of the document when running with docker is not good. There is not enough explanation.

**Reply by jertel** [reply]
What you've written above is a good quick start for Docker. Feel free to merge it into this page if you get a chance: https://elastalert2.readthedocs.io/en/latest/running_elastalert.html#as-a-docker-container.

**Reply by nsano-rururu** [reply]
PR
https://github.com/jertel/elastalert2/pull/623
